src/main/java/com/blamejared/crafttweaker/impl/tag/manager/TagManagerWrapper.java

```diff
@@ -0,0 +1,126 @@
+package com.blamejared.crafttweaker.impl.tag.manager;
+
+import com.blamejared.crafttweaker.api.*;
+import com.blamejared.crafttweaker.api.brackets.*;
+import com.blamejared.crafttweaker.impl.actions.tags.*;
+import com.blamejared.crafttweaker.impl.tag.*;
+import com.google.common.collect.*;
+import net.minecraft.tags.*;
+import net.minecraft.util.*;
+import net.minecraftforge.common.*;
+import net.minecraftforge.registries.*;
+
+import javax.annotation.*;
+import java.lang.invoke.*;
+import java.util.*;
+import java.util.stream.*;
+
+@ParametersAreNonnullByDefault
+public class TagManagerWrapper<T extends CommandStringDisplayable> implements TagManager<T> {
+    
+    @Nonnull
+    private final Class<T> elementClass;
+    @Nonnull
+    private final ResourceLocation tagTypeName;
+    @Nonnull
+    private final MethodHandle unwrapperHandle;
+    @Nonnull
+    private final MethodHandle wrapperHandle;
+    
+    public TagManagerWrapper(Class<T> elementClass, ResourceLocation tagTypeName, MethodHandle unwrapperHandle, MethodHandle wrapperHandle) {
+        this.elementClass = elementClass;
+        this.tagTypeName = tagTypeName;
+        this.unwrapperHandle = unwrapperHandle;
+        this.wrapperHandle = wrapperHandle;
+    }
+    
+    @Nonnull
+    @Override
+    public Class<T> getElementClass() {
+        return elementClass;
+    }
+    
+    public ITagCollection<?> getTagCollection() {
+        final Map<ResourceLocation, ITagCollection<?>> customTagTypes = ForgeTagHandler.getCustomTagTypes();
+        if(customTagTypes.containsKey(tagTypeName)) {
+            return customTagTypes.get(tagTypeName);
+        }
+        
+        if(TagRegistryManager.idToRegistryMap.containsKey(tagTypeName)) {
+            CraftTweakerAPI.logDebug("Falling back to TagRegistryManager for collection '" + tagTypeName + '\'');
+            return TagRegistryManager.idToRegistryMap.get(tagTypeName).getCollection();
+        }
```

Comment:
This will need to be checked/tested relatively strongly, as I am unsure timing wise if this will be grabbing the proper tag collection. The reason the JavaDocs for `ForgeTagHandler#getCustomTagTypes` says to prefer interacting via the `ITagCollectionSupplier` is to make sure that while a reload is happening the proper "intermediary" tag collection is what gets used, that will be set when reloading is complete. Given my guess is CrT happens sometime in the middle of this so potentially should be targeting the "new"" tag collection rather than the "old" one. The same goes I believe for the tag collections in the actual registries (though there is a slight chance those actually get updated more immediately).

The reason for this weird split from what I understand is so that if something goes wrong with datapacks, vanilla can safely roll back to the last successful loaded set.